### PR TITLE
[rum] guide on upgrade to browser sdk v3

### PR DESCRIPTION
**Pull request by hdelaby**
### What does this PR do?

Guide that summarises the evolutions of Browser SDK v3.

### Motivation

New feature and necessity to document breaking changes.

### Preview

https://docs-staging.datadoghq.com/hdelaby/migrate-v3-browser/real_user_monitoring/guide/v3-browser-sdk-upgrade/

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/real_user_monitoring/guide/_index.md [hunk 1]
```diff
@@ -8,5 +8,6 @@ private: true
     {{< nextlink href="real_user_monitoring/guide/send-custom-user-actions" >}}Send Custom User Actions{{< /nextlink >}}
     {{< nextlink href="real_user_monitoring/guide/upload-javascript-source-maps" >}}Upload Javascript source maps{{< /nextlink >}}
     {{< nextlink href="real_user_monitoring/guide/enrich-and-control-rum-data" >}}Enrich and control browser RUM data{{< /nextlink >}}
+    {{< nextlink href="real_user_monitoring/guide/v3-browser-sdk-upgrade" >}}Upgrading the Browser SDK to v3{{< /nextlink >}}
```
Comment (ruthnaebeck) [inline content/en/real_user_monitoring/guide/_index.md:11]:
```suggestion
    {{< nextlink href="real_user_monitoring/guide/browser-sdk-upgrade" >}}Upgrade the Browser SDK{{< /nextlink >}}
```

File: content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+---
+title: Upgrading the Browser SDK to v3
```
Comment (ruthnaebeck) [inline content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md:2]:
```suggestion
title: Upgrade the Browser SDK
```

File: content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+---
```
Comment (ruthnaebeck) [inline content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md:1]: Rename file to `browser-sdk-upgrade.md`

File: content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  tag: 'Documentation'
+  text: 'Visualize your RUM data in the Explorer'
+- link: "https://www.datadoghq.com/blog/session-replay-datadog/"
+  tag: "Blog"
+  text: "Use Datadog Session Replay to view real-time user journeys"
+---
+
+## Overview
+
+Version 3 of the RUM Browser SDK introduces [Session Replay][1]. With this new major version, several breaking changes were made to the Browser SDK.
```
Comment (ruthnaebeck) [inline content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md:15]:
```suggestion
RUM Browser SDK v3 introduces [Session Replay][1]. With this major version update, several breaking changes were made to the Browser SDK.
```

File: content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+### Stopping the generation of a RUM error for failed network requests
```
Comment (ruthnaebeck) [inline content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md:17]:
```suggestion
## Changes
### RUM errors
```

File: content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+The Browser SDK no longer issues [RUM errors][2] for failed XHR and Fetch calls. These failed network requests are still collected as [RUM resources][3], which contain the status code attribute.
+
+
+If you want to continue seeing these failed network requests as RUM errors, Datadog recommends intercepting the resource with the [beforeSend API][4], checking the `status_code` property and manually sending an error with the [addError API][5].
```
Comment (ruthnaebeck) [inline content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md:22]:
```suggestion
To continue seeing the failed network requests as RUM errors, Datadog recommends intercepting the resource with the [beforeSend API][4], checking the `status_code` property, and manually sending an error with the [addError API][5].
```

File: content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+```javascript
+beforeSend: (event) => {
+    if (event.type === 'resource' && event.resource.status_code >= 500) {
+        datadogRum.addError(`${event.resource.method} ${event.resource.url} ${event.resource.status_code}`); // "GET https://www.example.com/ 504"
+    }
+}
+```
+
+### Removing the source attribute from the addError API
```
Comment (ruthnaebeck) [inline content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md:32]:
```suggestion
### Source attribute
```

File: content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+The Browser SDK no longer lets you specify the source of an error collected with the [addError API][5]. All errors collected with this API now have their source attribute set to `custom`. The [addError API][5] accepts a context object as its second parameter, which should be used to pass extra context about the error.
```
Comment (ruthnaebeck) [inline content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md:34]:
```suggestion
The Browser SDK no longer lets you specify the source of an error collected with the [addError API][5]. All errors collected with this API have their source attribute set to `custom`. The [addError API][5] accepts a context object as its second parameter, which should be used to pass extra context about the error.
```

File: content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+### Deprecated Typescript types
```
Comment (ruthnaebeck) [inline content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md:36]:
```suggestion
## Deprecated
### TypeScript types
```

File: content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+| Old types                    | New types                    |
+| ---------------------------- | ---------------------------- |
+| RumUserConfiguration         | RumInitConfiguration         |
+| RumRecorderUserConfiguration | RumRecorderInitConfiguration |
+
+### Deprecated initialization options
```
Comment (ruthnaebeck) [inline content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md:43]:
```suggestion
### Initialization options
```

File: content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+
+| Old options        | New options |
+| ------------------ | ----------- |
+| publicApiKey       | clientToken |
+| datacenter         | site        |
+| resourceSampleRate | NONE        |
+
+### Deprecated API
```
Comment (ruthnaebeck) [inline content/en/real_user_monitoring/guide/v3-browser-sdk-upgrade.md:51]:
```suggestion
### API
```

**Review by ruthnaebeck — CHANGES_REQUESTED** [review]

**Comment by hdelaby** [comment]
Thank @ruthnaebeck, you took it to the next level 🙇 🥇  Ready to merge :)

**Review by bcaudan — APPROVED** [review]

**Review by ruthnaebeck — APPROVED** [review]

**Pull request merged**